### ISSUE TITLE
Do not build in debug mode under windows

### ISSUE THREAD

**Issue by kif** [mesonbuild/meson-python#187]
Hi,
While packaging (fabio) under windows using meson-python, I noticed the compilation fails at link time since it is willing to link to the debug version of the python library: 'python310_d.lib' (and this, despite the debug flag is turned off). I fount that passing the extra option `meson configure --buildtype plain` prevents the build to be performed in debug mode.
```
(py310) D:\kieffer\fabio>python -m build -n -x
* Building sdist...
+ meson setup --native-file=D:\kieffer\fabio\.mesonpy-native-file.ini -Ddebug=false -Doptimization=2 --prefix=c:\python\3.10 D:\kieffer\fabio D:\kieffer\fabio\.mesonpy-z6ihm2lw\build
The Meson build system
Version: 0.63.3
Source dir: D:\kieffer\fabio
Build dir: D:\kieffer\fabio\.mesonpy-z6ihm2lw\build
Build type: native build
Project name: FabIO
Project version: 0.15.0a1
Activating VS 16.7.6
C compiler for the host machine: cl (msvc 19.27.29112 "Microsoft (R) C/C++ Optimizing Compiler Version 19.27.29112 for x64")
C linker for the host machine: link link 14.27.29112.0
Cython compiler for the host machine: cython (cython 0.29.32)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program cython found: YES (D:\kieffer\py310\Scripts\cython.EXE)
Library m found: NO
Program python found: YES (D:\kieffer\py310\Scripts\python.exe)
Build targets in project: 6

FabIO 0.15.0a1

  User defined options
    Native files: D:\kieffer\fabio\.mesonpy-native-file.ini
    debug       : false
    optimization: 2
    prefix      : c:\python\3.10

Found ninja-1.10.2.git.kitware.jobserver-1 at D:\kieffer\py310\Scripts\ninja.EXE

Visual Studio environment is needed to run Ninja. It is recommended to use Meson wrapper:
D:\kieffer\py310\Scripts\meson compile -C .
+ meson dist --allow-dirty --no-tests --formats gztar
Activating VS 16.7.6
WARNING: Repository has uncommitted changes that will not be included in the dist tarball
Created D:\kieffer\fabio\.mesonpy-z6ihm2lw\build\meson-dist\FabIO-0.15.0a1.tar.gz
* Building wheel from sdist
* Building wheel...
+ meson setup --native-file=C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1\.mesonpy-native-file.ini -Ddebug=false -Doptimization=2 --prefix=c:\python\3.10 C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1 C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1\.mesonpy-wvagaqdx\build
The Meson build system
Version: 0.63.3
Source dir: C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1
Build dir: C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1\.mesonpy-wvagaqdx\build
Build type: native build
Project name: FabIO
Project version: 0.15.0a1
Activating VS 16.7.6
C compiler for the host machine: cl (msvc 19.27.29112 "Microsoft (R) C/C++ Optimizing Compiler Version 19.27.29112 for x64")
C linker for the host machine: link link 14.27.29112.0
Cython compiler for the host machine: cython (cython 0.29.32)
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program cython found: YES (D:\kieffer\py310\Scripts\cython.EXE)
Library m found: NO
Program python found: YES (D:\kieffer\py310\Scripts\python.exe)
Build targets in project: 6

FabIO 0.15.0a1

  User defined options
    Native files: C:\Users\kieffer\AppData\Local\Temp\build-via-sdist-opi8v7db\fabio-0.15.0a1\.mesonpy-native-file.ini
    debug       : false
    optimization: 2
    prefix      : c:\python\3.10

Found ninja-1.10.2.git.kitware.jobserver-1 at D:\kieffer\py310\Scripts\ninja.EXE

Visual Studio environment is needed to run Ninja. It is recommended to use Meson wrapper:
D:\kieffer\py310\Scripts\meson compile -C .
+ meson compile
Activating VS 16.7.6
[9/20] Linking target fabio/ext/cf_io.cp310-win_amd64.pyd
FAILED: fabio/ext/cf_io.cp310-win_amd64.pyd
"link"  /MACHINE:x64 /OUT:fabio/ext/cf_io.cp310-win_amd64.pyd fabio/ext/cf_io.cp310-win_amd64.pyd.p/meson-generated_fabio_ext_cf_io.pyx.c.obj fabio/ext/cf_io.cp310-win_amd64.pyd.p/src_columnfile.c.obj "/nologo" "/release" "/nologo" "/DLL" "/IMPLIB:fabio\ext\cf_io.cp310-win_amd64.lib" "c:\python\3.10\libs\python310.lib" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
LINK : fatal error LNK1104: cannot open file 'python310_d.lib'
[14/20] Compiling C object fabio/ext/mar345_IO.cp310-win_amd64.pyd.p/meson-generated_fabio_ext_mar345_IO.pyx.c.obj
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "D:\kieffer\py310\lib\site-packages\pep517\in_process\_in_process.py", line 351, in <module>
    main()
  File "D:\kieffer\py310\lib\site-packages\pep517\in_process\_in_process.py", line 333, in main
    json_out['return_val'] = hook(**hook_input['kwargs'])
  File "D:\kieffer\py310\lib\site-packages\pep517\in_process\_in_process.py", line 249, in build_wheel
    return _build_backend().build_wheel(wheel_directory, config_settings,
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 987, in build_wheel
    return project.wheel(out).name
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 935, in wheel
    wheel = self._wheel_builder.build(self._build_dir)
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 576, in build
    self._project.build()  # ensure project is built
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 765, in build
    self._meson('compile')
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 696, in _meson
    return self._proc('meson', *args)
  File "D:\kieffer\py310\lib\site-packages\mesonpy\__init__.py", line 691, in _proc
    subprocess.check_call(list(args))
  File "c:\python\3.10\lib\subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['meson', 'compile']' returned non-zero exit status 1.

ERROR Backend subprocess exited when trying to invoke build_wheel
```

**Comment by rgommers** [comment]
@kif is this issue fixed by https://github.com/mesonbuild/meson/pull/10800?

**Comment by dnicolodi** [comment]
This must be fixed, because the CI for meson-python and my projects says that building on Windows works.

**Comment by kif** [comment]
> @kif is this issue fixed by [mesonbuild/meson#10800](https://github.com/mesonbuild/meson/pull/10800)?

Yep, fabio is not built with meson, also under windows.
